server/src/main/java/org/opensearch/action/admin/indices/replication/TransportSegmentReplicationStatsAction.java

```diff
@@ -181,4 +200,78 @@ protected ClusterBlockException checkRequestBlock(
     ) {
         return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_READ, concreteIndices);
     }
+
+    private SegmentReplicationShardStats calcualteSegmentReplicationShardStats(ShardRouting shardRouting) {
+        ShardId shardId = shardRouting.shardId();
+        SegmentReplicationState completedSegmentReplicationState = targetService.getlatestCompletedEventSegmentReplicationState(shardId);
```

Comment:
I don't think we want to compare these state objects to compute these stats, rather they should be computed from the ReplicationCheckpoint currently served by the shard and the one on the state?

Reply:
As we discussed offline, we will not be comparing with the last completed checkpoint. Instead of this we are going to calculate Bytes Remaining To Replicate. This can be calculated using the `FileMetadata` details in the `ReplicationLuceneIndex` in the `SegmentReplicationState`.